source/_docs/cookies.md

```diff
@@ -0,0 +1,62 @@
+---
+title: Working with Cookies on Pantheon
+description: Learn tips and tricks for working with cookies.
+categories: [developing]
+tags: [varnish]
+keywords: varnish, cache, clear cache, caching, drupal, wordpress, cookies, cookie, setcookie, set-cookie
+---
+
+## Disable Caching for Specific Pages
+You can use regular expression(s) to determine if the current request (`$_SERVER['REQUEST_URI']`) should be excluded from cache. If the request matches, bypass cache by setting the `NO_CACHE` cookie in the response:
+```
+//Set or replace $regex_path_match accordingly
+if ((preg_match($regex_path_match, $_SERVER['REQUEST_URI'])) {
+  $domain =  $_SERVER['HTTP_HOST'];
+  setcookie('NO_CACHE', '1', time()+0, '/path-to-page', $domain);
+}
+```
+As an alternative to setting a `NO_CACHE` cookie within the response, you can [modify the `Cache-Control:` header](/docs/cache-control) to bypass cache on Pantheon.
+
+## Cache-varying Cookies
+Respond to a request with cached content depending on the presence and value of a particular cookie. It's important to note that in order for the response to be cached by Pantheon's edge, the cookie name must match `STYXKEY[a-zA-Z0-9_-]`.
+
+First, check to see if the cookie is set within the incoming request. If the cookie is set, store the value and use it to generate varied content as appropriate for your use case and implementation.
+
+<div class="alert alert-info" role="alert">
+<h4>Note</h4>
+If the value has already been set, do not set the cookie again in the response. Varnish cannot cache a response that contains a <code>Set-Cookie:</code> header.
+</div>
+
+If the value is **not** set, respond with `setcookie()` to serve cached content for subsequent requests within the defined cookie lifetime.
+
+The following example can be used interchangeably between WordPress and Drupal:
+```
+if(isset($_COOKIE['STYXKEY_gorp'])){
+
+  $foo = $_COOKIE['STYXKEY_gorp'];
```

Comment:
I don't fully understand this code block, why do we assign the value of the cookie? I don't immediately see how it gets used later.